src/PowerFlowData.jl

```diff
@@ -40,44 +131,114 @@ function PowerFlowData(::ACPowerFlow, sys::PSY.System)
         bus = PSY.get_component(PSY.Bus, sys, bus_name)
         bus_type[ix] = PSY.get_bustype(bus)
         bus_angles[ix] = PSY.get_angle(bus)
+        # ! shouldn't it be like this?
+        # ! if bus_type[ix] == PSY.BusTypes.REF
+        # !    bus_angles[ix] = 0.0
+        # ! else
+        # !     bus_angles[ix] = PSY.get_angle(bus)
+        # ! end
         bus_magnitude[ix] = PSY.get_magnitude(bus)
     end
 
     bus_activepower_injection = zeros(Float64, n_buses)
     bus_reactivepower_injection = zeros(Float64, n_buses)
-    get_injections!(bus_activepower_injection, bus_reactivepower_injection, bus_lookup, sys)
+    _get_injections!(
+        bus_activepower_injection,
+        bus_reactivepower_injection,
+        bus_lookup,
+        sys,
+    )
 
     bus_activepower_withdrawals = zeros(Float64, n_buses)
     bus_reactivepower_withdrawals = zeros(Float64, n_buses)
-    get_withdrawals!(
+    _get_withdrawals!(
         bus_activepower_withdrawals,
         bus_reactivepower_withdrawals,
         bus_lookup,
         sys,
     )
 
+    # initialize data
+    init_1 = zeros(n_buses, timesteps)
+    init_2 = zeros(n_branches, timesteps)
+
+    # define fields as matrices whose number of columns is eqault to the number of timesteps
+    bus_activepower_injection_1 = deepcopy(init_1)
```

Comment:
using 

```suggestion
    bus_activepower_injection_1 = similar(init_1)
```

might be faster than copies

Reply:
done